.github/workflows/ci.yaml

```diff
@@ -86,60 +51,156 @@ jobs:
-      
+          echo "::set-output name=starttime_test::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[0].started_at')" && echo "::set-output name=endtime_test::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[0].completed_at')"
+          echo "::set-output name=starttime_build::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[1].started_at')" && echo "::set-output name=endtime_build::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs | jq '.jobs[1].completed_at')"
+
       - name: calculate time
         id: calculate_time
         run: |
-          started_at=${{steps.execution_timestamps.outputs.starttime}}
-          completed_at=${{steps.execution_timestamps.outputs.endtime}}
+          started_at_test=${{steps.execution_timestamps.outputs.starttime_test}}
+          completed_at_test=${{steps.execution_timestamps.outputs.endtime_test}}
+          started_at_build=${{steps.execution_timestamps.outputs.starttime_build}}
+          completed_at_build=${{steps.execution_timestamps.outputs.endtime_build}}
 
-          START_TIME=$(date -d "$started_at" +%s)
-          END_TIME=$(date -d "$completed_at" +%s)
+          START_TIME_TEST=$(date -d "$started_at_test" +%s)
+          END_TIME_TEST=$(date -d "$completed_at_test" +%s)
+          START_TIME_BUILD=$(date -d "$started_at_build" +%s)
+          END_TIME_BUILD=$(date -d "$completed_at_build" +%s)
 
-          ELAPSE=$(( $END_TIME - $START_TIME ))
+          ELAPSE_TEST=$(( $END_TIME_TEST - $START_TIME_TEST ))
+          ELAPSE_BUILD=$(( $END_TIME_BUILD - $START_TIME_BUILD ))
+          ELAPSE=$(( $ELAPSE_TEST + $ELAPSE_BUILD ))
           echo $ELAPSE
           echo "$(($ELAPSE/60/60))h $(($ELAPSE/60%60))m $(($ELAPSE%60))s"
           echo "::set-output name=elpsd::$ELAPSE"
           echo "::set-output name=el_hours::$(($ELAPSE/60/60))"
           echo "::set-output name=el_minutes::$(($ELAPSE/60%60))"
           echo "::set-output name=el_seconds::$(($ELAPSE%60))"
+      
+      - name: get email for Committer on PR or pusher on PUSH
+        id: email
+        run: |
+          if [[ ${{ github.event_name }} == 'pull_request' ]]; then
+            echo "::set-output name=user_email::$(curl -H "Accept: application/vnd.github+json" https://api.github.com/repos/${{ github.repository }}/actions/runs | jq '.workflow_runs[0].head_commit.committer.email')"
+          else
+            echo "::set-output name=user_email::${{ github.event.pusher.email }}"
```

Comment:
find how to get committer email using github env vars github actions make available to you. No need to call API.

Reply:
haven't found environment variable for committer email on pull request no matter how much I've tried. Using ' git log -1 --pretty=format:%ae '